packages/react/src/components/MenuButton/MenuButton.tsx

```diff
@@ -0,0 +1,331 @@
+    this.trySetState({
+      open,
+      autoFocus: true,
+    })
+  }
+
+  handleMenuItemClick = (predefinedProps?: MenuProps) => (
+    e: React.SyntheticEvent,
+    itemProps: MenuItemProps,
+  ) => {
+    _.invoke(predefinedProps, 'onItemClick', e, itemProps)
+    if (!itemProps || !itemProps.menu) {
+      // do not close if clicked on item with submenu
+      this.trySetState({ open: false, autoFocus: false })
+    }
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    unhandledProps,
+    accessibility,
+    styles,
+  }: RenderResultConfig<MenuButtonProps>): React.ReactNode {
+    const {
+      // MenuButton props:
+      contextMenu,
+      menu,
+      // Popup props:
+      accessibility: accessibilityProp,
+      align,
+      className,
+      defaultOpen,
+      inline,
+      mountDocument,
+      mountNode,
+      mouseLeaveDelay,
+      offset,
+      on,
+      onOpenChange,
+      open,
+      pointing,
+      position,
+      tabbableTrigger,
+      styles: stylesProp,
+      target,
+      trigger,
+      unstable_pinned,
+      variables,
+    } = this.props
+
+    const popupProps = _.omitBy(
+      {
+        accessibility: accessibilityProp,
+        align,
+        className,
+        defaultOpen,
+        inline,
+        mountDocument,
+        mountNode,
+        mouseLeaveDelay,
+        offset,
+        on,
+        onOpenChange,
+        open,
+        pointing,
+        position,
+        tabbableTrigger,
+        styles: stylesProp,
+        target,
+        trigger,
+        unstable_pinned,
+        variables,
+      },
+      _.isUndefined,
+    )
+
+    const content = Menu.create(menu, {
+      defaultProps: {
+        ...accessibility.attributes.menu,
+        vertical: true,
+      },
+      overrideProps: {
+        onItemClick: (e: React.SyntheticEvent, itemProps: MenuItemProps) =>
+          this.handleMenuItemClick(content)(e, itemProps),
+      },
+    })
+
+    const overrideProps = {
+      open: this.state.open,
+      onOpenChange: this.handleOpenChange,
+      content: {
+        styles: styles.popupContent,
+        content: content && <Ref innerRef={this.menuRef}>{content}</Ref>,
+      },
+      children: undefined, // force-reset `children` defined for `Popup` as it collides with the `trigger
+    }
+
+    if (contextMenu) {
+      return Popup.create(popupProps, {
+        defaultProps: {
+          on: 'context',
+          trapFocus: true,
+          unstable_pinned: true,
+          tabbableTrigger: false,
+        },
+        overrideProps,
+      })
+    }
+
+    return (
+      <ElementType
+        className={classes.root}
+        {...accessibility.attributes.root}
+        {...unhandledProps}
+        {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
+      >
+        <Ref innerRef={this.triggerRef}>
+          {Popup.create(popupProps, {
+            defaultProps: {
+              inline: true,
+              autoFocus: true,
```

Comment:
Please test this, should be `this.state.autoFocus` or `this.state.autoFocus` should be removed